django/template/defaultfilters.py

```diff
@@ -134,10 +134,7 @@ def floatformat(text, arg=-1):
     if not m and p < 0:
         return mark_safe(formats.number_format('%d' % (int(d)), 0))
 
-    if p == 0:
-        exp = Decimal(1)
-    else:
-        exp = Decimal('1.0') / (Decimal(10) ** abs(p))
+    exp = Decimal(10) ** -abs(p)
```

Comment:
Did you consider `decimal.Decimal(1).scaleb(-abs(p))`? It seems that it could be faster when `p != 0`:
```
In [81]: %timeit decimal.Decimal(10) ** 0
347 ns ± 7.74 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [82]: %timeit decimal.Decimal(1).scaleb(0)
470 ns ± 3.89 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

```
In [83]: %timeit decimal.Decimal(10) ** -1
661 ns ± 5.87 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [84]: %timeit decimal.Decimal(1).scaleb(-1)
481 ns ± 4.66 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
@sir-sigurd I didn't. Thanks for tip :rocket: , we can do the same in https://github.com/django/django/pull/9468.

Reply:
Yes, will fix it.